index.js

```diff
@@ -603,160 +604,133 @@ class LatticeKeyring extends EventEmitter {
   // [re]connect to the Lattice. This should be done frequently to ensure
   // the expected wallet UID is still the one active in the Lattice.
   // This will handle SafeCard insertion/removal events.
-  _connect() {
-    return new Promise((resolve, reject) => {
+  async _connect () {
+    try {
       // Attempt to connect with a Lattice using a shorter timeout. If
       // the device is unplugged it will time out and we don't need to wait
       // 2 minutes for that to happen.
       this.sdkSession.timeout = CONNECT_TIMEOUT;
-      this.sdkSession.connect(this.creds.deviceID, (err) => {
-        this.sdkSession.timeout = SDK_TIMEOUT;
-        if (err) {
-          return reject(err);
-        }
-        if (!this._syncCurrentWalletUID()) {
-          return reject('No active wallet.');
-        }
-        return resolve();
-      });
-    })
+      await this.sdkSession.connect(this.creds.deviceID)
+      this.sdkSession.timeout = SDK_TIMEOUT;
+    } catch (err) {
+      this.sdkSession.timeout = SDK_TIMEOUT;
+      throw new Error(err);
+    }
   }
 
-  _initSession() {
-    return new Promise((resolve, reject) => {
-      if (this.isUnlocked()) {
-        return resolve();
-      }
-      try {
-        let url = 'https://signing.gridpl.us';
-        if (this.creds.endpoint)
-          url = this.creds.endpoint
-        let setupData;
-        if (this.sdkState) {
-          // If we have state data we can fully rehydrate the session.
-          setupData = {
-            stateData: this.sdkState
-          }
-        } else {
-          // If we have no state data, we need to create a session.
-          // Its state will be saved once the connection is established.
-          setupData = {
-            name: this.appName,
-            baseUrl: url,
-            timeout: SDK_TIMEOUT,
-            privKey: this._genSessionKey(),
-            network: this.network,
-          }
-        }
-        this.sdkSession = new SDK.Client(setupData);
-        // Return a boolean indicating whether we provided state data.
-        // If we have, we can skip `connect`.
-        return resolve(!!setupData.stateData);
-      } catch (err) {
-        return reject(err);
-      }
-    })
+  // Force state to sync with that of the Lattice so we know the wallet UID
+  async _syncWallet() {
```

Comment:
This could also be written: 

```js
_syncWallet() {
  return this.sdkSession
  	.fetchActiveWallet()
  	.catch(() => this._connect())
}
```

I find it a bit easier to parse, but that might be my bias.

Reply:
Going to leave it as is. This function will be deprecated in the next version anyway